doc/modules/ROOT/pages/python-runtime/knowledge-graph-embeddings.adoc

```diff
@@ -0,0 +1,296 @@
+When negative sampling size is more than one, the loss is computed for a positive triple and each of its negative triples, and the average loss is computed.
+
+image::python-runtime/mrl.svg[width=300]
+image::python-runtime/delta-value.svg[width=400]
+
+
+==== Negative Sampling Self-Adversarial Loss
+
+Negative Sampling Self-Adversarial Lossfootnote:[Sun, Zhiqing, et al. "Rotate: Knowledge graph embedding by relational rotation in complex space." arXiv preprint arXiv:1902.10197 (2019).] is a setwise loss function that compares the scores based on the difference between the scores of a positive triple and a set of negative triples.
+`loss_function_kwargs` can be used to set the `adversarial_temperature` and `margin` parameters.
+
+
+=== Optimizer
+
+Any pytorch optimizer can be used for training the model.
+To use non-default optimizer, specify the optimizer class name as a string in the `optimizer` parameter.
+All optimizer parameters except `params` can be passed as `optimizer_kwargs`.
+
+
+=== Negative sampling
+
+The loss function requires negative samples to compute the loss.
+The number of negative samples per positive sample is controlled by the `negative_sampling_size` parameter.
+When `use_node_type_aware_sampler` is set to `True`, negative nodes are sampled with the same label as the corresponding positive node.
+With or without node type awareness, the negative samples are sampled uniformly at random from the graph.
+
+
+=== Learning rate scheduler
+
+Any pytorch learning rate scheduler can be used for training the model.
+To use non-default learning rate scheduler, specify the scheduler class name as a string in the `lr_scheduler` parameter.
+All scheduler parameters except `optimizer` can be passed as `lr_scheduler_kwargs`.
+
+
+=== Inner normalisation
+
+In the original `TransE` paperfootnote:[Bordes, Antoine, et al. "Translating embeddings for modeling multi-relational data." Advances in neural information processing systems 26 (2013).]
+in `Algorithm 1`, line 5, the entity embeddings are normalized to have `Lp` norm of 1.
+Value of `p` is set by the `p_norm` parameter.
+For some datasets, this normalization might not be beneficial.
+To avoid this normalization, set `inner_norm` to `False`.
+
+
+=== Filtered metrics
+
+When we evaluate (compute metrics) the model, we compute scores for all possible triples with the same head or tail and relationship type as the test triple.
+Ranking the test triple among other triples is used for computing metrics, such as Mean Rank, Mean Reciprocal Rank, and Hit@k.
+
+When `filtered_metrics` is set to `False`, the ranking is done among all possible triples.
+
+When `filtered_metrics` is set to `True`, the ranking is done among only the triples that are not present in the training set.
+
+[[algorithms-embeddings-kge-syntax]]
+== Syntax
+
+[source, python, role=no-test]
+----
+gds.model.train(G,
```

Comment:
I don't think that `model` is sufficiently clear, as there are several types of models in the library. Did you consider `kge` or something instead? Or perhaps something else

Reply:
Yes, I think we have to mention `kge`.